rotkehlchen/db/utils.py

```diff
@@ -405,3 +405,38 @@ def get_query_chunks(data: Sequence[int | str]) -> list[tuple[Sequence[int | str
         ((chunk := data[i:i + SQL_VARIABLE_CHUNK_SIZE]), ','.join(['?'] * len(chunk)))
         for i in range(0, len(data), SQL_VARIABLE_CHUNK_SIZE)
     ]
+
+
+def unlock_database(
+        db_connection: 'DBConnection',
+        password: str,
+        sqlcipher_version: int,
+        apply_optimizations: bool = True,
+) -> None:
+    """Unlock a SQLCipher encrypted database connection.
+
+    Applies the decryption key and sets up standard pragmas for rotkehlchen databases.
```

Comment:
explain what apply_optimizations does. Also the name seems wrong. It also activates foreign keys if that is True. That is not an optimization but a very fundamental DB setting which shuld be on in almost all of the cases.